File: 3274/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(checkTwoChessboards("a1", "c3"))
	fmt.Println(checkTwoChessboards("a1", "h3"))
	fmt.Println(checkTwoChessboards("a7", "a6"))	// false
}

func checkTwoChessboards(coordinate1, coordinate2 string) bool {
	cs1 := coordinate1[0] - 'a' + 1 + coordinate1[1] - '0'
	cs2 := coordinate2[0] - 'a' + 1 + coordinate2[1] - '0'

	return cs1 % 2 == cs2 % 2
}

File: 3274/main.rs
struct Solution{}

impl Solution {
    pub fn check_two_chessboards(coordinate1: String, coordinate2: String) -> bool {
        (coordinate1.chars().nth(0).unwrap() as i32 - coordinate2.chars().nth(0).unwrap() as i32
        + coordinate1.chars().nth(1).unwrap() as i32 - coordinate2.chars().nth(1).unwrap() as i32) % 2 == 0
    }
}

fn main() {
    println!("{}", Solution::check_two_chessboards("a7".to_string(), "a6".to_string()));
}